rule/rule.go

```diff
@@ -903,3 +912,17 @@ func (s byAttrName) Less(i, j int) bool {
 func (s byAttrName) Swap(i, j int) {
 	s[i], s[j] = s[j], s[i]
 }
+
+func checkFile(f *File) error {
+	names := make(map[string]bool)
+	for _, r := range f.Rules {
+		if r.Name() == "" {
```

Comment:
Let's extract `r.Name()` into a variable. `rule.Name` is not a simple getter, so we should avoid calling it multiple times if we can.